### ISSUE TITLE
[Local testing] copy_into_table raises NotImplementedError, but it appears in list of supported APIs

### ISSUE THREAD

**Issue by mraraujo** [snowflakedb/snowpark-python#1620]
Using snowflake-snowpark-python==1.16.0.

_analyzer.py checks

```
        if isinstance(logical_plan, CopyIntoTableNode):
            self._conn.log_not_supported_error(
                external_feature_name="DateFrame.copy_into_table",
                raise_error=NotImplementedError,
            )
```

However, DataFrame.copy_into_table appears explicitly in the list of supported APIs for local testing in: https://docs.snowflake.com/en/developer-guide/snowpark/python/testing-locally#list-of-supported-apis

**Comment by sfc-gh-sghosh** [comment]
Hello @mraraujo ,

Thanks for raising the issue. The data frame.copy_into_table() API is not yet supported, so we will fix the documentation.

Regards,
Sujan

**Comment by mraraujo** [comment]
@sfc-gh-aling yes, we use copy_into_table to ingest data into snowflake due to its flexibility (especially the ability to define behavior on_error and to log the results of the copy).
Adding it to local testing would help us increase test coverage on those parts of our system as well.

**Comment by sfc-gh-aling** [comment]
thanks @mraraujo for the context, we will prioritize this feature request!
